From d154815d387bff5fe66687528ecd93f35571a81e Mon Sep 17 00:00:00 2001
From: effects3d <effects3dee@gmail.com>
Date: Sun, 27 Sep 2026 13:46:54 +0200
Subject: [PATCH 17/19] winex11: Don't hide cursor under X11 sessions

Avoids issues with the cursor locking under X11 sessions. Requires
"winex11: Use XFixes to hide cursor before warping it." patch.

Signed-off-by: effects3d <effects3dee@gmail.com>
---
 dlls/winex11.drv/x11drv.h      | 1 +
 dlls/winex11.drv/x11drv_main.c | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/dlls/winex11.drv/x11drv.h b/dlls/winex11.drv/x11drv.h
index 2b7e919..d40aedb 100644
--- a/dlls/winex11.drv/x11drv.h
+++ b/dlls/winex11.drv/x11drv.h
@@ -487,6 +487,7 @@ extern BOOL use_primary_selection;
 extern BOOL use_system_cursors;
 extern BOOL grab_fullscreen;
 extern BOOL usexcomposite;
+extern BOOL xwayland;
 extern BOOL use_xfixes;
 extern BOOL managed_mode;
 extern BOOL private_color_map;
diff --git a/dlls/winex11.drv/x11drv_main.c b/dlls/winex11.drv/x11drv_main.c
index 5c0d2f7..9c181af 100644
--- a/dlls/winex11.drv/x11drv_main.c
+++ b/dlls/winex11.drv/x11drv_main.c
@@ -69,6 +69,7 @@ BOOL usexvidmode = TRUE;
 BOOL usexrandr = TRUE;
 BOOL usexcomposite = TRUE;
 BOOL use_egl = TRUE;
+BOOL xwayland = FALSE;
 BOOL use_xfixes = FALSE;
 BOOL use_take_focus = TRUE;
 BOOL use_primary_selection = FALSE;
@@ -729,6 +730,8 @@ NTSTATUS __wine_unix_lib_init(void)
                    DisplayHeight( display, default_visual.screen ));
     X11DRV_Settings_Init();
 
+    xwayland = !strcmp(getenv("XDG_SESSION_TYPE"), "wayland");
+
     /* initialize XVidMode */
     X11DRV_XF86VM_Init();
     /* initialize XRandR */
-- 
2.56.0

